client/my-sites/stats/pages/realtime/chart.tsx

```diff
@@ -0,0 +1,126 @@
+import moment from 'moment';
+import { useEffect, useState, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import AsyncLoad from 'calypso/components/async-load';
+import wpcom from 'calypso/lib/wp';
+import { getSiteOption } from 'calypso/state/sites/selectors';
+import { parseChartData } from 'calypso/state/stats/lists/utils';
+import PageLoading from '../shared/page-loading';
+
+type Unit = 'hour' | 'day' | 'week' | 'month' | 'year';
+
+interface QueryStatsVisitsParams {
+	unit: Unit;
+	date: string;
+	quantity: number;
+	stat_fields: string;
+}
+
+interface chartMinuteDataTypes {
+	[ key: string ]: number;
+}
+
+function queryStatsVisits( siteId: number, params: QueryStatsVisitsParams ) {
+	return wpcom.req.get( `/sites/${ siteId }/stats/visits`, params );
+}
+// TODO: Change to 60 for production.
+const UPDATE_INTERVAL_IN_SECONDS = 5;
+const MINUTE_DATA_LENGTH = 30;
+
+const RealtimeChart = ( { siteId }: { siteId: number } ) => {
+	const gmtOffset = useSelector( ( state: object ) =>
+		getSiteOption( state, siteId, 'gmt_offset' )
+	) as number;
+	const [ viewsData, setViewsData ] = useState( {} as chartMinuteDataTypes );
+
+	useEffect( () => {
+		const intervalId = setInterval( () => {
+			// Index the chart data by YYYY-MM-DD HH:mm:00.
+			const adjustedDatetime = moment().utcOffset( gmtOffset ).format( 'YYYY-MM-DD HH:mm:00' );
+
+			queryStatsVisits( siteId, {
+				unit: 'hour',
+				date: adjustedDatetime,
+				quantity: 1,
+				stat_fields: 'views',
+			} ).then( ( response: any ) => {
+				const result = parseChartData( response );
+				const views = result[ 0 ].views || 0;
+
+				setViewsData( ( prevViewsData ) => {
+					return {
+						...prevViewsData,
+						[ adjustedDatetime ]: views,
+					};
+				} );
+			} );
+		}, UPDATE_INTERVAL_IN_SECONDS * 1000 );
+
+		return () => clearInterval( intervalId );
+	}, [ siteId, gmtOffset ] );
+
+	let chartData = useMemo( () => {
+		return Object.keys( viewsData ).map( ( eachMinute, idx ) => {
+			const lastMinute = moment( eachMinute )
+				.subtract( 1, 'minute' )
+				.format( 'YYYY-MM-DD HH:mm:00' );
+
+			// Reset the data if the current minute is not continuous with the previous minute,
```

Comment:
Does it reset all the data on the chart? It'll be great if we could resume and continue to show metric after resuming on the page I guess.